src/x_connective.c

```diff
@@ -1015,63 +1026,104 @@ static void trigger_list(t_trigger *x, t_symbol *s, int argc, t_atom *argv)
+    for(i = x->x_n, u = x->x_vec + i; u--, i--;)
     {
         if (u->u_type == TR_FLOAT)
             outlet_float(u->u_outlet, (argc ? atom_getfloat(argv) : 0));
         else if (u->u_type == TR_BANG)
             outlet_bang(u->u_outlet);
         else if (u->u_type == TR_SYMBOL)
             outlet_symbol(u->u_outlet,
-                (argc ? atom_getsymbol(argv) : &s_symbol));
+                          (argc ? atom_getsymbol(argv) : (s != NULL ? s : &s_symbol)));
+        else if (u->u_type == TR_ANYTHING)
+            outlet_anything(u->u_outlet, s, argc, argv);
         else if (u->u_type == TR_POINTER)
         {
             if (!argc || argv->a_type != TR_POINTER)
                 pd_error(x, "unpack: bad pointer");
             else outlet_pointer(u->u_outlet, argv->a_w.w_gpointer);
         }
+        else if (u->u_type == TR_STATIC_FLOAT)
+            outlet_float(u->u_outlet, u->u_float);
         else outlet_list(u->u_outlet, &s_list, argc, argv);
     }
 }
 
 static void trigger_anything(t_trigger *x, t_symbol *s, int argc, t_atom *argv)
 {
+    t_atom *av2 = NULL;
     t_triggerout *u;
-    int i;
-    for (i = (int)x->x_n, u = x->x_vec + i; u--, i--;)
+    int i, j = 0;
+    for (i = x->x_n, u = x->x_vec + i; u--, i--;)
     {
         if (u->u_type == TR_BANG)
             outlet_bang(u->u_outlet);
         else if (u->u_type == TR_ANYTHING)
             outlet_anything(u->u_outlet, s, argc, argv);
-        else pd_error(x, "trigger: can only convert 's' to 'b' or 'a'");
+        else if (u->u_type == TR_STATIC_FLOAT)
+            outlet_float(u->u_outlet, u->u_float);
+        else
+        {
+            // copying trigger_list behavior except that here we keep
+            // the outlet number and therefore avoid redundant printouts
+            if (u->u_type == TR_FLOAT)
+                outlet_float(u->u_outlet, (argc ? atom_getfloat(argv) : 0));
+            else if (u->u_type == TR_BANG)
+                outlet_bang(u->u_outlet);
+            else if (u->u_type == TR_SYMBOL)
+                outlet_symbol(u->u_outlet,
+                              (s != NULL ? s : (argc ? atom_getsymbol(argv) : &s_symbol)));
+            else if (u->u_type == TR_ANYTHING)
+                outlet_anything(u->u_outlet, s, argc, argv);
+            else if (u->u_type == TR_POINTER)
+                if (!argc || argv->a_type != TR_POINTER)
+                    pd_error(x, "unpack: bad pointer");
+                else outlet_pointer(u->u_outlet, argv->a_w.w_gpointer);
+                else if (u->u_type == TR_STATIC_FLOAT)
+                    outlet_float(u->u_outlet, u->u_float);
+                else
+                { // Ico: don't have to worry about zero element case (AFAICT)
+                    av2 = (t_atom *)getbytes((argc + 1) * sizeof(t_atom));
+                    SETSYMBOL(av2, s);
+                    if (argc == 0)
+                        outlet_list(u->u_outlet, &s_symbol, argc+1, av2);
+                    else
```

Comment:
the code path in the else clause is all we need (as Ico has pointed out in his comment), so you can remove the if check for `arg == 0`.

Reply:
done